### PR TITLE
Implemented: Service for sending generic request (#1tkuhhw)

### PR DESCRIPTION
**Pull request by sheetalj2205**
*  Created a service for sending requests 
*  This service take three in-params of shippingGatewayConfigId, path and requestMap and give responseMap as out param.

Related Ticket Link: [https://app.clickup.com/t/1up4u0m]

### CODE REVIEW

**Comment by sheetalj2205** [comment]
@dixitdeepak Please review this PR.

File: service/mantle/ups/UpsServices.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        </out-parameters>
+        <actions>
+            <entity-find-one entity-name="ups.ShippingGatewayUps" value-field="upsShippingConfig" cache="true"/>
+            <if condition="upsShippingConfig == null || !(upsShippingConfig.userId || upsShippingConfig.password)">
+                <log message="No shipping configuration found for upsShippingConfigId ${upsShippingConfigId} "/>
+                <return type="warning" error="true"
+                        message="No shipping configuration found for upsShippingConfigId ${upsShippingConfigId}"/>
+            </if>
+            <set field="responseMap" from="[:]"/>
+            <set field="requestMap" from="[:]"/>
```
Comment (dixitdeepak) [inline service/mantle/ups/UpsServices.xml]: why you are initializing it again, its part of in parameters, so it should not override here.

File: service/mantle/ups/UpsServices.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                  import org.moqui.util.RestClient;
+                  import groovy.json.JsonSlurper;
+
+                  def connectUrl = upsShippingConfig.connectUrl;
+                  if (!connectUrl.endsWith("/")) {
+                    connectUrl+="/";
+                    }
+                 connectUrl = connectUrl + path;
+                 RestClient restClient = ec.service.rest()
+                 restClient.method("POST")
```
Comment (dixitdeepak) [inline service/mantle/ups/UpsServices.xml]: Are we sure all method will be post?
As this is generic service so it should support all kind of method. We should pass this as service in param.

File: service/mantle/ups/UpsServices.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                 restClient.contentType("application/json")
+                 restClient.jsonObject(requestMap)
+                 RestClient.RestResponse restResponse = restClient.call()
+                 if (restResponse.statusCode < 200 || restResponse.statusCode > 300) {
+                    errMsgMap = (Map) new JsonSlurper().parseText(restResponse.text())
+                    ec.logger.warn(restResponse.text())
+                    if (errMsgMap?.error) {
+                        responseMap.put("error", errMsgMap?.error?.message)
+                    } else if (errMsgMap?.errors) {
+                        responseMap.put("error", errMsgMap?.title)
```
Comment (dixitdeepak) [inline service/mantle/ups/UpsServices.xml]: please check other way to return all the error message, there are method/reference example in moqui framework to prepare message from list.

File: service/mantle/ups/UpsServices.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                  if (!connectUrl.endsWith("/")) {
+                    connectUrl+="/";
+                    }
+                 connectUrl = connectUrl + path;
+                 RestClient restClient = ec.service.rest()
+                 restClient.method("POST")
+                 restClient.uri(connectUrl)
+                 restClient.contentType("application/json")
+                 restClient.jsonObject(requestMap)
+                 RestClient.RestResponse restResponse = restClient.call()
```
Comment (dixitdeepak) [inline service/mantle/ups/UpsServices.xml:37]: Need to handle the authentication part.

**Comment by sheetalj2205** [comment]
@dixitdeepak Please check

File: .gitignore [hunk 5]
```diff
@@ -1 +1,3 @@
+=======
```
Comment (dixitdeepak) [inline .gitignore]: Please remove this

**Comment by sheetalj2205** [comment]
@dixitdeepak Please merge this PR now. Resolved all issues.

**Review by dixitdeepak — APPROVED** [review]

**Pull request merged**